### PR TITLE
e2e test on kubernetes v1.22.1

### PR DESCRIPTION
**Pull request by STRRL**
Signed-off-by: STRRL <str_ruiling@outlook.com>

### What problem does this PR solve?

Issue Number: https://github.com/chaos-mesh/chaos-mesh/issues/2172#issuecomment-918011219

### What is changed and how it works?

What's Changed:

- add kubernetes v1.22.1 on test matrix

### Related changes

* PR to update `chaos-mesh/website`/`chaos-mesh/website-zh`: No.
* Need to update Chaos Dashboard component, related issue: No.
* Need to cheery-pick to the release branch: No.

### Checklist

Tests

- [ ] Unit test
- [x] E2E test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

### Release note 

```release-note
Please add a release note.
If you don't think this PR needs a release note then fill it with None.
```

### CODE REVIEW

**Comment by STRRL** [comment]
e2e failed because of the rate limit from docker.io, trying to fix that.

File: ci/pingcap_chaos_mesh_build_kind.groovy [hunk 1]
```diff
@@ -38,6 +38,13 @@ spec:
     env:
     - name: DOCKER_IN_DOCKER_ENABLED
       value: "true"
+    - name: DOCKER_IO_MIRROR
+      value: https://registry-mirror.pingcap.net
+    - name: DOCKER_IO_MIRROR
```
Comment (Colstuwjx) [inline ci/pingcap_chaos_mesh_build_kind.groovy:43]: Duplicated?

Comment (STRRL) [reply ci/pingcap_chaos_mesh_build_kind.groovy:43]: Oh, that should be `GCR_IO_MIRROR`. 🤣

Comment (STRRL) [reply ci/pingcap_chaos_mesh_build_kind.groovy]: fixed!

**Comment by STRRL** [comment]
PTAL

/cc @Colstuwjx 
/cc @YangKeao

File: ci/pingcap_chaos_mesh_build_kind.groovy [hunk 2]
```diff
@@ -250,8 +257,11 @@ def call(BUILD_BRANCH, CREDENTIALS_ID) {
 		builds["E2E on kubernetes 1.12.10"] = {
                 build("v1.12", "${GLOBALS} GINKGO_NODES=6 KUBE_VERSION=v1.12.10 KIND_VERSION=0.8.1 ./hack/e2e.sh -- --ginkgo.focus='Basic'")
         }
-        builds["E2E on kubernetes 1.20.4"] = {
+        builds["E2E on kubernetes 1.20.2"] = {
                 build("v1.20", "${GLOBALS} GINKGO_NODES=6 KUBE_VERSION=v1.20.2 ./hack/e2e.sh -- --ginkgo.focus='Basic'")
+        }
+        builds["E2E on kubernetes 1.22.1"] = {
+                build("v1.22", "${GLOBALS} GINKGO_NODES=6 KUBE_VERSION=v1.22.1 ./hack/e2e.sh -- --ginkgo.focus='Basic'")
```
Comment (Colstuwjx) [inline ci/pingcap_chaos_mesh_build_kind.groovy:264]: For a long-term plan, we may need a document to announce the official support k8s version list of chaos mesh, and run e2e tests against these versions.

BTW, after we added this version, how many extra time cost has been introduced? From the [jenkins dashboard](https://ci.pingcap.net/blue/organizations/jenkins/chaos-mesh-e2e/activity), it looks like not a big problem 😄

**Review by YangKeao — APPROVED** [review]
LGTM

File: ci/pingcap_chaos_mesh_build_kind.groovy (hunk 2 above)
Comment (STRRL) [reply ci/pingcap_chaos_mesh_build_kind.groovy:264]: > For a long-term plan, we may need a document to announce the official support k8s version list of chaos mesh, and run e2e tests against these versions.

We are considering making a table about that, we are looking around for the similar things that other kubernetes operators do. :heart:

> BTW, after we added this version, how many extra time cost has been introduced? From the jenkins dashboard, it looks like not a big problem 😄

We execute each test in parallel, jenkins will create different workloads on different jenkins-worker(as different pods), so the time would not increase.

**Review by Colstuwjx — APPROVED** [review]
LGTM!

**Pull request merged**